plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -1981,6 +2081,45 @@ public void testPartitionPredicatePushdownWithHistoricalPartitionSpecs()
         dropTable(tableName);
     }
 
+    @Test
+    public void testPartitionPredicatePushdownWithNestedFieldPartitioning()
+    {
+        // Start with a bucket transform, which cannot be used for predicate pushdown
+        String tableName = "test_partition_predicate_pushdown_with_nested_field_partitioning";
+        assertUpdate("CREATE TABLE " + tableName + " (parent ROW(child1 TIMESTAMP(6), child2 INTEGER)) WITH (partitioning = ARRAY['bucket(\"parent.child2\", 3)'])");
+        String selectQuery = "SELECT parent.child2 FROM " + tableName + " WHERE CAST(parent.child1 AS date) < DATE '2015-01-02'";
+
+        String initialValues =
+                "ROW(ROW(TIMESTAMP '1969-12-31 22:22:22.222222', 8))," +
+                        "ROW(ROW(TIMESTAMP '1969-12-31 23:33:11.456789', 9))," +
+                        "ROW(ROW(TIMESTAMP '1969-12-31 23:44:55.567890', 10))";
+        assertUpdate("INSERT INTO " + tableName + " VALUES " + initialValues, 3);
+        assertThat(query(selectQuery))
+                .containsAll("VALUES 8, 9, 10")
+                .isNotFullyPushedDown(FilterNode.class);
+
+        String hourTransformValues =
+                "ROW(ROW(TIMESTAMP '2015-01-01 10:01:23.123456', 1))," +
+                        "ROW(ROW(TIMESTAMP '2015-01-02 10:10:02.987654', 2))," +
+                        "ROW(ROW(TIMESTAMP '2015-01-03 10:55:00.456789', 3))";
+        // While the bucket transform is still used the hour transform still cannot be used for pushdown
```

Comment:
// While the bucket transform is still used, the hour transform cannot be used for pushdown